docs-ref-conceptual/use-cli-effectively.md

```diff
@@ -213,6 +213,9 @@ If you only need create or update commands, use the [az deployment group create]
 
 If generic update arguments and [az resource](../latest/docs-ref-autogen/resource.yml) don't meet your needs, you can use [az rest](/cli/azure/reference-index#az-res) command to call the REST API. The command automatically authenticates using the logged-in credential and sets header `Content-Type: application/json`. For more information, see [Azure REST API reference](/rest/api/azure/).
 
+> [!NOTE]
+> Calling service API directly with `az rest` is only a supplemental feature of Azure CLI. It is not guaranteed to work and is subject to change, for reasons including but not limited to Azure CLI permission change, Microsoft Graph service limitation, AAD Conditional Access policies, etc. For service API questions or issues, please [create a support ticket](/support/create-ticket/).
```

Comment:
> we don't have "supplemental" commands. We only have experimental, preview, and GA. `az rest` is actually `core` b/c there is no extension warning, but I can't tell the status from the [autogenerated ref content](https://docs.microsoft.com/en-us/cli/azure/reference-index?view=azure-cli-latest#az-rest-examples).

`az rest` is GA, but its usage is too generic and a user can use it to call any API, even incorrectly.

> desired result

The difficult question is how we define "desired result". In https://github.com/Azure/azure-cli/issues/22903, it's not that `az rest` is not producing the desired result, but the user is calling the API **incorrectly**.

Reply:
@jiasli , how does a customer know what queries are supported by `az rest`?  Instead of trying to identify all the correct and incorrect variations, is there a consistent error message?

"
Note! Not all queries are supported by `az rest` for reasons including, but not limited to, Azure CLI permission changes, Microsoft Graph service limitations, AAD Conditional Access policies, etc."  If your query is not supported by `az rest`, you will receive the following error: .....
"

I am trying to convey that this command does work if used correctly vs "not guaranteed to work"; make sense?

Reply:
> how does a customer know what queries are supported by `az rest`? Instead of trying to identify all the correct and incorrect variations, is there a consistent error message?

No. This totally depends on the API itself. A customer will need to refer to the API doc for supported queries, behaviors, error messages, permissions required, etc.

However, there are many limitations, sometimes because Azure CLI is forbidden to do that, sometimes because the user is forbidden to do that. That's why this command is "not guaranteed to work". Yes, I admit some more precise wording is needed here. 🤔